### PR TITLE
MFB-928: WA - Utility Discount Program (UDP)

### PR DESCRIPTION
**Pull request by david-j-stoll**
## Context & Motivation

- Closes MFB-928
- Adds a custom `ProgramCalculator` for the Seattle Utility Discount Program (UDP), which provides discounts of up to 60% on Seattle City Light electric bills and 50% on Seattle Public Utilities water/sewer/garbage bills for income-qualified households.

## Changes Made

- **`programs/programs/wa/udp/calculator.py`** — New `WaUdp` calculator. Eligibility requires residence in the SCL/SPU service area (proxied via 31 Seattle ZIP codes + King County) AND one of three financial pathways: SSI categorical, SNAP streamlined, or adult (18+) household income ≤ 70% Washington State Median Income. Minor income is excluded per DPP III-428 §5.6.2. Value is $732/year (average annual savings per seattle.gov/human-services).
- **`programs/programs/wa/__init__.py`** — Registered `WaUdp` under key `"wa_udp"` in `wa_calculators`.
- **`programs/programs/wa/udp/tests/test_udp.py`** — 30 unit tests covering: class attributes, SMI limit table + extension logic, location criterion, income pathway (including boundary conditions), adult-only income aggregation, SSI categorical bypass, SNAP streamlined bypass, and benefit value.
- **`programs/programs/wa/udp/spec.md`** — Research spec with full eligibility criteria, benefit value methodology, data gaps, and 10 test scenarios.
- **`programs/management/commands/import_program_config_data/data/wa_udp_initial_config.json`** — Initial program config (metadata, documents, navigators).
- **`validations/management/commands/import_validations/data/wa_udp.json`** — 3 validation scenarios: standard income path (eligible), income $1 over HH=4 limit (ineligible), SSI categorical bypass (eligible).
- **`configuration/white_labels/wa.py`** — Added `udp` to the `housingAndUtilities` category in `category_benefits`.
- **`screener/models.py`** — Added `has_udp` BooleanField to `Screen`; mapped `"udp"` and `"wa_udp"` in `has_benefit()`.
- **`screener/serializers.py`** — Added `has_udp` to `ScreenSerializer` fields.
- **`screener/migrations/0153_screen_has_udp.py`** — Migration for `has_udp` field.

## Testing

- Migrations to run:
  - `python manage.py migrate screener`
- Configuration updates needed:
  - `python manage.py import_program_config programs/management/commands/import_program_config_data/data/wa_udp_initial_config.json`
  - `python manage.py import_validations validations/management/commands/import_validations/data/wa_udp.json`
- Environment variables/settings to add: none
- Manual testing steps:
  1. Run unit tests: `pytest programs/programs/wa/udp/tests/test_udp.py -v`
  2. Activate the program in the Django admin (`Program.active = True` for `wa_udp`)
  3. Run validations: `python manage.py validate -w wa -p wa_udp` — all 3 should pass
  4. Run the full WA validation suite: `python manage.py validate -w wa` — confirm no regressions

## Deployment

- Post-deployment scripts needed:
  - `python manage.py migrate screener` (adds `has_udp` column)
  - `python manage.py import_program_config programs/management/commands/import_program_config_data/data/wa_udp_initial_config.json`
  - `python manage.py import_validations validations/management/commands/import_validations/data/wa_udp.json`
- Production config updates: none
- Admin updates needed: set `Program.active = True` for `wa_udp` when ready to go live
- Notify team/users of: new program available on the WA white label for Seattle households

## Notes for Reviewers

- Known limitations:
  - **Account ownership data gap**: The legal eligibility test requires a Seattle City Light or Seattle Public Utilities bill in the applicant's name (or Voucher Program enrollment). The calculator proxies this with ZIP code + King County — non-customers who pass the filter are caught at application time.
  - **SMI table is hard-coded for 2026**: WA DSHS publishes updated figures each January. The table in the calculator will need to be updated annually (same pattern as other WSOS calculators).
  - **Monthly vs. annual income tests**: The spec requires both tests, but they collapse to the same check at the screener level since the limits are exactly proportional and the screener captures ongoing income rates rather than a 1-month lookback snapshot.
  - **Landlord-paid utility credit path**: The screener can't identify tenant households whose landlord participates in the UDP Voucher Program. All eligible households get the $732/year average value regardless.
- Future considerations:
  - Adding a utility provider field to the screener (suggested in spec) would close the account-ownership data gap and also benefit future WA utility programs (PSE, Snohomish PUD, Tacoma, etc.).

## Summary by CodeRabbit

* **New Features**
  * Added support for the Seattle Utility Discount Program (WA UDP) with Seattle residency checks, three eligibility pathways (income test, SSI categorical, SNAP streamlined), and a default annual benefit value.

* **Documentation**
  * Added a comprehensive WA UDP specification covering criteria, exclusions, benefit methodology, and test scenarios.

* **Tests**
  * Added unit tests covering eligibility pathways, location rules, income boundaries, and benefit value.

* **Chores**
  * Added initial program configuration and test fixture data for WA UDP.

[Review Change Stack](https://app.coderabbit.ai/change-stack/MyFriendBen/benefits-api/pull/1536?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: programs/programs/wa/udp/calculator.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        3: 82_740,
+        4: 98_508,
+        5: 114_264,
+        6: 130_032,
+        7: 132_984,
+        8: 135_936,
+        9: 138_900,
+        10: 141_852,
+    }
+    SMI_70_PER_EXTRA_ANNUAL = 2_964  # +$2,964/year per person above HH=10
```
[the remark this answers is not in the document]
Comment (david-j-stoll) [reply programs/programs/wa/udp/calculator.py:37]: Resolving, the values are already calculating 70% of the median income as evidenced by this chart: https://www.seattle.gov/human-services/services-and-programs/utility-discount-program

Compared to the dshs website: https://www.dshs.wa.gov/esa/eligibility-z-manual-ea-z/state-median-income-chart. The each additional appears to be accurate after comparing to that chart.

File: configuration/white_labels/wa.py [hunk 2]
```diff
@@ -1196,6 +1196,16 @@ def get_white_label(self) -> WhiteLabel:
...
+                "udp": {
+                    "name": {
+                        "_label": "housingAndUtilities.udp",
+                        "_default_message": "Seattle Utility Discount Program (UDP): ",
+                    },
+                    "description": {
+                        "_label": "housingAndUtilities.udp_desc",
+                        "_default_message": "Discounts on Seattle utility bills",
+                    },
+                },
```
Comment (catonph) [inline configuration/white_labels/wa.py]: Don't worry about adding this - it did inform the "what benefits do you already have" page but we are halfway through updating that page to use a separate join table

File: programs/programs/wa/udp/calculator.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            if member.age is not None and member.age >= 18
+        )
+
+    def _has_ssi_recipient(self) -> bool:
+        return any(member.calc_gross_income("yearly", ["sSI"]) > 0 for member in self.screen.household_members.all())
+
+    def household_eligible(self, e: Eligibility) -> None:
+        # Criterion 1: Seattle SCL/SPU service area (ZIP + county proxy)
+        in_seattle = self.screen.zipcode in self.SEATTLE_ZIP_CODES and self.screen.county == "King County"
+        e.condition(in_seattle, messages.location())
```
Comment (catonph) [inline programs/programs/wa/udp/calculator.py:123]: Let's put county restriction via the admin portal - or is that not looking specific enough?

File: programs/programs/wa/udp/calculator.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        size = self.screen.household_size
+        if size in self.SMI_70_ANNUAL:
+            return self.SMI_70_ANNUAL[size]
+        return self.SMI_70_ANNUAL[10] + (size - 10) * self.SMI_70_PER_EXTRA_ANNUAL
+
+    def _adult_income_yearly(self) -> float:
+        return sum(
+            member.calc_gross_income("yearly", ["all"])
+            for member in self.screen.household_members.all()
+            if member.age is not None and member.age >= 18
```
Comment (catonph) [inline programs/programs/wa/udp/calculator.py]: Let's use `member.calc_age()` here. It honors `screen.get_reference_date()` and falls back to `member.age` when `birth_year_month` is missing, which keeps things consistent with validation fixtures

File: programs/programs/wa/udp/calculator.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    are exactly proportional and the screener captures ongoing income rates, not a 1-month snapshot.
+
+    Data gaps: account ownership and meter sharing (mixed-load exclusion) cannot be verified at
+    screening. Calculator assumes neither disqualifying condition applies (inclusivity assumption).
+    """
+
+    amount = 732  # average annual savings per seattle.gov/human-services UDP page
+
+    # 2026 70% WA State Median Income annual limits — WA DSHS, effective January 1, 2026
+    SMI_70_ANNUAL: ClassVar[dict[int, int]] = {
```
Comment (catonph) [inline programs/programs/wa/udp/calculator.py:25]: I hate that we hardcode this but alas this is our precedent - we need a better system at some point

**Review by catonph — CHANGES_REQUESTED** [review]
A couple small things but otherwise looks good!

File: programs/programs/wa/udp/calculator.py (hunk 3 above)
Comment (david-j-stoll) [reply programs/programs/wa/udp/calculator.py:123]: I was wondering about that too. The postal codes here seem to be truly defining Seattle proper. If we restrict to King County via the admin portal then I believe the additional check on the postal code will make the calculator work as expected. Correct me if I'm wrong there.

File: configuration/white_labels/wa.py (hunk 2 above)
Comment (david-j-stoll) [reply configuration/white_labels/wa.py]: You're saying we don't need to update this file at all? Sounds good, will revert this change.

**Review by catonph — APPROVED** [review]
Just update based on our convo in slack about zip codes! otherwise looks good to go
